test/framework/src/main/java/org/elasticsearch/test/AbstractQueryTestCase.java

```diff
@@ -330,6 +349,10 @@ public void onRemoval(ShardId shardId, Accountable accountable) {
         namedWriteableRegistry = injector.getInstance(NamedWriteableRegistry.class);
     }
 
+    protected static void registerPlugin(Class<? extends Plugin> pluginClass) {
```

Comment:
I just spoke to @cbuescher and it has a perf impact. I think we should move that entire thing into a Holder class that we pull in and only create once and put it in a static member ie. something like this:

```Java

@Before
public void setUp() {
  super.setUp();
  if (holderReference == null) { // holderReference is static
    holderReference = new Holder(plugins);
  }
}

@AfterClass
public void afterClass() {
  holderReference = null;
}
```